extensions/wikia/InsightsV2/InsightsLoopController.class.php

```diff
@@ -0,0 +1,201 @@
+	 */
+	private function setMustacheParams( $params, $isFixed, $type ) {
+		$html = MustacheService::getInstance()->render(
+			'extensions/wikia/InsightsV2/templates/Insights_loopNotification.mustache',
+			$params
+		);
+
+		$this->response->setData([
+			'html' => $html,
+			'isFixed' => $isFixed,
+			'notificationType' => $type
+		]);
+	}
+
+
+	/**
+	 * Get params for notification template shown in edit mode
+	 *
+	 * @param string $subpage Insights subpage
+	 * @param InsightsQueryPageModel $model
+	 * @return array
+	 */
+	private function getInProgressNotificationParams( $subpage, InsightsQueryPageModel $model ) {
+		$params = $this->getInsightListLinkParams( $subpage );
+		$params['notificationMessage'] = wfMessage( InsightsHelper::INSIGHT_INPROGRESS_MSG_PREFIX . $subpage )->plain();
+
+		if ( $model->getConfig()->displayFix() ) {
+			$params['fixItMessage'] = wfMessage( 'insights-notification-message-fixit' )->plain();
+		}
+
+		return $params;
+	}
+
+	/**
+	 * Get params for notification template shownwhen user dont fix an issue
+	 *
+	 * @param string $subpage Insights subpage
+	 * @param Title $title
+	 * @param InsightsQueryPageModel $model
+	 * @return array
+	 */
+	private function getNotFixedNotificationParams( $subpage, Title $title, InsightsQueryPageModel $model ) {
+		$params = $this->getInsightListLinkParams( $subpage );
+
+		if ( $model->getConfig()->displayFix() ) {
+			$params = array_merge( $params, $this->getInsightFixItParams( $title, $model ) );
+		}
+
+		$params['notificationMessage'] = wfMessage( InsightsHelper::INSIGHT_INPROGRESS_MSG_PREFIX . $subpage )->plain();
+
+		return $params;
+	}
+
+	/**
+	 * Get params for notification template shown when user fix all issues in given insight type
+	 */
+	private function getCongratulationsNotificationParams() {
+		$params = $this->getInsightLinkParams();
+		$params['notificationMessage'] = wfMessage( 'insights-notification-message-alldone' )->plain();
+
+		return $params;
+	}
+
+	/**
+	 * Get params for notification template shown when user fix one issue from the insights list
+	 *
+	 * @param string $subpage Insights subpage
+	 * @param array $next An array with a url and text for a link to the next item
+	 * @return array
+	 */
+	private function getInsightFixedNotificationParams( $subpage, $next ) {
+		$params = $this->getInsightNextLinkParams( $subpage, $next );
+		$params = array_merge( $params, $this->getInsightListLinkParams( $subpage ));
+		$params['notificationMessage'] = wfMessage( InsightsHelper::INSIGHT_FIXED_MSG_PREFIX . $subpage )->plain();
+
+		return $params;
+	}
+
+	/**
+	 * Get params to generate next item link in notification template
+	 *
+	 * @param String $subpage Insights subpage
+	 * @param array $next An array with a url and text for a link to the next item
+	 * @return array
+	 */
+	private function getInsightNextLinkParams( $subpage, $next ) {
+		return [
+			'nextArticleText' => wfMessage( 'insights-notification-next-item-' . $subpage )->plain(),
```

Comment:
For consistency, this prefix should probably also have a constant in the helper class.